php/libraries/Database.class.inc

```diff
@@ -765,6 +765,28 @@ class Database
     }
 
     /**
+     * Runs a query as a prepared statement and returns the values of the first
+     * column in the select statement. If multiple columns are given, only the first
+     * will be returned.
+     *
+     * @param string $query  The SQL SELECT query to be run
+     * @param array  $params Values to use for binding to prepared statement
+     *
+     * @return array Associative array of form rowID=>value containing all values
+     *               for the first element of the select statement
+     */
+    function pselectCol($query, $params)
+    {
+        $unprocessed = $this->pselect($query, $params);
+
+        $result = array();
+        foreach ($unprocessed as $k=>$row) {
```

Comment:
This should probably ensure `count($row) === 1`, and throw an exception if the query had the wrong number of columns.

Reply:
It could also use a comment explaining what "reset" does, I had to look it up.